test/unit/role_lock_test.rb

```diff
@@ -0,0 +1,101 @@
+require 'test_helper'
+
+class RoleLockTest < ActiveSupport::TestCase
+  setup do
+    @role_lock = Foreman::Plugin::RoleLock.new('foreman_test')
+    @permissions = [:view_hosts]
+  end
+
+  def find_role(name)
+    Role.find_by :name => name
+  end
+
+  test "should rename role with a free name" do
+    role = roles(:create_hosts)
+    role_name = role.name
+    prefix = "Customized"
+    new_name = "#{prefix} #{role_name} 1"
+    refute find_role(new_name)
+    @role_lock.rename_with_free_name role, prefix, role_name
+    assert find_role(new_name)
+  end
+
+  test "should rename existing role" do
+    role = roles(:create_hosts)
+    original_name = role.name
+    new_name = "Customized #{original_name}"
+    refute find_role(new_name)
+    @role_lock.rename_existing role, original_name
+    assert find_role(new_name)
+  end
+
+  test "should create plugin role" do
+    name = "Test Manager"
+    refute find_role(name)
+    @role_lock.create_plugin_role name, @permissions
+    role = find_role(name)
+    assert role
+    assert_equal @permissions, role.permissions.pluck(:name).map(&:to_sym)
+    assert_equal @role_lock.plugin_id, role.origin
+  end
+
+  test "should rename and create role" do
+    original = roles(:create_hosts)
+    original_id = original.id
+    original_name = original.name
+    new_name = "Customized #{original_name}"
+    refute find_role(new_name)
+    @role_lock.rename_and_create original, original_name, @permissions
+    renamed_original = find_role(new_name)
+    assert renamed_original
+    assert_equal original_id, renamed_original.id
+    new_role = find_role(original_name)
+    assert new_role
+    assert_equal @permissions, new_role.permissions.pluck(:name).map(&:to_sym)
+    refute renamed_original.locked?
+    assert_equal @role_lock.plugin_id, new_role.origin
+  end
+
+  test "should process existing role with changed permissions list" do
+    role = roles(:create_hosts)
+    original_name = role.name
+    original_id = role.id
+    @role_lock.process_role original_name, @permissions
+    renamed_role = find_role("Customized Create hosts")
+    assert renamed_role
+    assert_equal original_id, renamed_role.id
+    new_role = find_role(original_name)
+    assert new_role
+    refute_equal renamed_role, new_role
+  end
+
+  test "should process existing role with unchanged permissions list" do
+    role = roles(:create_hosts)
+    original_name = role.name
+    original_id = role.id
+    refute role.locked?
+    @role_lock.process_role original_name, role.permissions.pluck(:name).map(&:to_sym)
+    updated_original = find_role(original_name)
+    assert_equal original_id, updated_original.id
+    assert_equal original_name, updated_original.name
+    assert @role_lock.plugin_id, updated_original.origin
```

Comment:
I'd add assert role.locked? just to be sure esp. since we checked this on line 76 with refute